plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestReusableConnectionFactory.java

```diff
@@ -139,6 +139,28 @@ public void testConnectionIsNotReusedWhenSetToReadOnly()
         }
     }
 
+    @Test
+    public void testConnectionIsNotReusedWhenDelegateIsClosed()
```

Comment:
I would remove comments. It looks like you repeat java code but in English. Also this scares a bit because first impression is that it is more complex than it is actually